Defunct email addresses cause storage cron to fail

It seems that Mailchimp does its email sending synchronously, and we don't catch the `EmailSendError`, so one failing email addresses causes the whole cron job to fail.

```shell output
📧 🚨 Failed to send User90PercentStorage email to REDACTED-NAME (REDACTED-NAME@REDACTED-DOMAIN.com) using mailchimp.
file:///home/runner/work/web3.storage/web3.storage/packages/cron/src/lib/email/providers/mailchimp.js:67
      throw new EmailSendError(`📧 🚨 Mailchimp failed to send. Status: ${response.status}, reason: ${response.reject_reason}.`)
            ^
EmailSendError: 📧 🚨 Mailchimp failed to send. Status: rejected, reason: hard-bounce.
    at MailchimpEmailProvider.sendEmail (file:///home/runner/work/web3.storage/web3.storage/packages/cron/src/lib/email/providers/mailchimp.js:67:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async EmailService.sendEmail (file:///home/runner/work/web3.storage/web3.storage/packages/cron/src/lib/email/service.js:76:19)
    at async checkStorageUsed (file:///home/runner/work/web3.storage/web3.storage/packages/cron/src/jobs/storage.js:145:27)
    at async main (file:///home/runner/work/web3.storage/web3.storage/packages/cron/src/bin/storage.js:14:5)
npm ERR! Lifecycle script `start:storage` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @web3-storage/cron@1.2.0 
npm ERR!   at location: /home/runner/work/web3.storage/web3.storage/packages/cron 
Error: Process completed with exit code 1.
```

--- a/packages/cron/src/jobs/storage.js
+++ b/packages/cron/src/jobs/storage.js
@@ -143,7 +143,8 @@ export async function checkStorageUsed ({ roPg, emailService, userBatchSize = 20
 
       if (emailToSend) {
         const emailSent = await emailService.sendEmail(to, emailToSend.emailType, {
-          ...(emailToSend.secondsSinceLastSent && { secondsSinceLastSent: emailToSend.secondsSinceLastSent })
+          ...(emailToSend.secondsSinceLastSent && { secondsSinceLastSent: emailToSend.secondsSinceLastSent }),
+          failSilently: true
         })
 
         if (emailToSend.emailType === EMAIL_TYPE.User100PercentStorage) {

--- a/packages/cron/test/storage.spec.js
+++ b/packages/cron/test/storage.spec.js
@@ -8,7 +8,9 @@ import { getDBClient, getPg } from '../src/lib/utils.js'
 import { EMAIL_TYPE } from '@web3-storage/db'
 import { checkStorageUsed } from '../src/jobs/storage.js'
 import { EmailService, EMAIL_PROVIDERS } from '../src/lib/email/service.js'
+import DummyEmailProvider from '../src/lib/email/providers/dummy.js'
 import sinon from 'sinon'
+import { EmailSendError } from '../src/lib/email/errors.js'
 
 const env = {
   ...process.env,
@@ -172,5 +174,16 @@ describe('cron - check user storage quotas', () => {
     assert.strictEqual(emailUsers75Calls[0].args[0].email, 'test2@email.com')
     assert.strictEqual(emailUsers75Calls[0].args[1], 'User75PercentStorage', 'user weekly check over 75')
     assert.strictEqual(emailUsers75Calls[0].args[2].secondsSinceLastSent, 60 * 60 * 24 * 7 - (60 * 60))
+  })
+
+  it('handles email send errors', async () => {
+    const providerSendEmailStub = sinon.stub(emailService.provider, 'sendEmail')
+    providerSendEmailStub.onFirstCall().throws(new EmailSendError())
+    providerSendEmailStub.returns('message ID 123')
+    await checkStorageUsed({ roPg, emailService })
+    assert.ok(
+      providerSendEmailStub.called,
+      'cron should have tried to send an email using its email provider'
+    )
   })
 })

